shell/browser/ui/win/electron_desktop_window_tree_host_win.cc

```diff
@@ -66,24 +64,20 @@ bool ElectronDesktopWindowTreeHostWin::GetDwmFrameInsetsInPixels(
 bool ElectronDesktopWindowTreeHostWin::GetClientAreaInsets(
     gfx::Insets* insets,
     HMONITOR monitor) const {
+  // Windows by deafult extends the maximized window slightly larger than
+  // current workspace, for frameless window since the standard frame has been
+  // removed, the client area would then be drew outside current workspace.
+  //
+  // Indenting the client area can fix this behavior.
   if (IsMaximized() && !native_window_view_->has_frame()) {
-    if (base::win::GetVersion() < base::win::Version::WIN8) {
-      // This tells Windows that most of the window is a client area, meaning
-      // Chrome will draw it. Windows still fills in the glass bits because of
-      // the // DwmExtendFrameIntoClientArea call in |UpdateDWMFrame|.
-      // Without this 1 pixel offset on the right and bottom:
-      //   * windows paint in a more standard way, and
-      //   * we get weird black bars at the top when maximized in multiple
-      //     monitor configurations.
-      int border_thickness = 1;
-      insets->Set(0, 0, border_thickness, border_thickness);
-    } else {
-      const int frame_thickness = ui::GetFrameThickness(monitor);
-      // Reduce the Windows non-client border size because we extend the border
-      // into our client area in UpdateDWMFrame(). The top inset must be 0 or
-      // else Windows will draw a full native titlebar outside the client area.
-      insets->Set(0, frame_thickness, frame_thickness, frame_thickness);
-    }
+    // The insets would be eventually passed to WM_NCCALCSIZE, which takes
+    // the metrics under the DPI of _main_ monitor instead of current moniotr.
```

Comment:
surprising find, should this be reported to upstream ?

Reply:
This only seems to be a problem for maximized frameless window, which Chrome only uses for deprecated Chrome Apps APIs. I can't find other way to reproduce it in Chrome .